sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -1659,7 +1659,8 @@ class Analyzer(
       // child of it.
       var currentChild = child
       var i = 0
-      while (i < groupedWindowExpressions.size) {
+      val size = groupedWindowExpressions.size
+      while (i < size) {
```

Comment:
In cases like this, can we use `fold` or an equivalent functional idiom to make this even simpler and possibly more efficient?